save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/Requests.kt

```diff
@@ -0,0 +1,123 @@
+/**
+ * Utilities to perform requests to other services of save-cloud
+ */
+
+package com.saveourtool.save.agent
+
+import com.saveourtool.save.agent.utils.*
+import com.saveourtool.save.agent.utils.extractZipTo
+import com.saveourtool.save.agent.utils.markAsExecutable
+import com.saveourtool.save.agent.utils.requiredEnv
+import com.saveourtool.save.agent.utils.unzipIfRequired
+import com.saveourtool.save.agent.utils.writeToFile
+import com.saveourtool.save.core.logging.logWarn
+import com.saveourtool.save.core.utils.runIf
+import com.saveourtool.save.domain.FileKey
+
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.onDownload
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import okio.Path
+import okio.Path.Companion.toPath
+
+/**
+ * Download test source snapshots for execution [executionId] into [target]
+ *
+ * @param config
+ * @param target
+ * @param executionId
+ * @return result
+ */
+internal suspend fun SaveAgent.downloadTestResources(config: BackendConfig, target: Path, executionId: String): Result<Unit> = runCatching {
+    val result = httpClient.downloadTestResources(config, executionId)
+    if (updateState(result)) {
+        return@runCatching
+    }
+
+    val response = result.getOrThrow()
+    val bytes = response.body<ByteArray>().runIf({ isEmpty() }) {
```

Comment:
May be extract into some method? similar logic is used in `downloadAdditionalResources`